### PR TITLE
Add pipline to trigger generating docs in internal repo

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes
This pr add the script to trigger generating docs and PRs in the internal repo.
A successful [job](https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android-internal/1520/workflows/61aa6987-2d79-44e1-a733-873e8d5a9144/jobs/3920) triggered from local verifies the result.

### CODE REVIEW

File: scripts/generate-docs-start-pipeline.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        data["branch"] = branch
+
+    r = requests.post(url, auth=(token, ""), headers=headers, json=data)
+
+    if r.status_code != 201 and r.status_code != 200:
+        print("Error triggering the CircleCI: %s." % r.json()["message"])
+        sys.exit(1)
+
+
+def Main():
```
Comment (alexander-kulikovskii) [inline scripts/generate-docs-start-pipeline.py:33]:
```suggestion
def main():
```

https://www.datacamp.com/community/tutorials/pep8-tutorial-python-code

File: scripts/generate-docs-start-pipeline.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+# Implements https://circleci.com/docs/api/v2/#trigger-a-new-pipeline
+
+import argparse
+import os
+import requests
+import sys
+
+
+def TriggerPipeline(slug, token, branch, params):
```
Comment (alexander-kulikovskii) [inline scripts/generate-docs-start-pipeline.py:11]:
```suggestion
def triggerPipeline(slug, token, branch, params):
```

File: scripts/generate-docs-start-pipeline.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        sys.exit(1)
+
+    params = {
+        "mapbox_android_upstream": True,
+        "mapbox_slug": args.origin_slug,
+        "mapbox_android_hash": args.hash,
+        "mapbox_release_tag": args.release_tag
+    }
+
+    TriggerPipeline(args.target_slug, args.token, args.branch, params)
```
Comment (alexander-kulikovskii) [inline scripts/generate-docs-start-pipeline.py:68]:
```suggestion
    TriggerPipeline(slug = args.target_slug, token = args.token, branch = args.branch, params = params)
```
https://treyhunner.com/2018/04/keyword-arguments-in-python/

File: scripts/generate-docs-start-pipeline.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    parser.add_argument("--hash", default=hash,
+            help="Commit git hash that triggered the pipeline, otherwise environment CIRCLE_SHA1.")
+    parser.add_argument("--branch",
+            help="Build a specific branch, otherwise it will build the default branch.")
+    parser.add_argument("--release_tag",
+            help="The release tag name, must provide.")
+
+    args = parser.parse_args()
+
+    if not args.token:
```
Comment (alexander-kulikovskii) [inline scripts/generate-docs-start-pipeline.py:57]:
```suggestion
    if not args.token or args.token.strip():
```
If I set token as spaces `"   "`

File: scripts/generate-docs-start-pipeline.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            help="CircleCI token, otherwise environment CIRCLE_API_TOKEN.")
+    parser.add_argument("--origin-slug", default="mapbox/mapbox-maps-android",
+            help="Origin repository, otherwise CIRCLE_PROJECT_USERNAME/CIRCLE_PROJECT_REPONAME.")
+    parser.add_argument("--target-slug", default="".join((username, '/', project)),
+            help="Repository to trigger the pipeline, example: mapbox/mapbox-gl-native-android.")
+    parser.add_argument("--hash", default=hash,
+            help="Commit git hash that triggered the pipeline, otherwise environment CIRCLE_SHA1.")
+    parser.add_argument("--branch",
+            help="Build a specific branch, otherwise it will build the default branch.")
+    parser.add_argument("--release_tag",
```
Comment (alexander-kulikovskii) [inline scripts/generate-docs-start-pipeline.py:52]: should this argument be required?

```suggestion
    parser.add_argument("--release_tag", required=True,
```

**Review by alexander-kulikovskii — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

**Pull request merged**